src/Feature/Registration/Feature_Registration.re

```diff
@@ -0,0 +1,406 @@
+      height(25),
+      width(Constants.macWidth),
+      justifyContent(`Center),
+      alignItems(`Center),
+      flexGrow(0),
+      paddingHorizontal(8),
+    ];
+  };
+};
+
+module View = {
+  open Revery;
+  open Revery.UI;
+
+  module Colors = Feature_Theme.Colors;
+  module FontIcon = Oni_Components.FontIcon;
+  module FontAwesome = Oni_Components.FontAwesome;
+
+  module Modal = {
+    let modal = (~children, ~theme, ()) =>
+      <View style=Styles.boxShadow>
+        <View style={Styles.container(~theme)}>
+          <View style=Style.[padding(5)]> children </View>
+        </View>
+      </View>;
+
+    let loadingAnimation =
+      Animation.(animate(Time.seconds(1)) |> tween(0., 6.28) |> repeat);
+
+    let%component make =
+                  (
+                    ~theme: ColorTheme.Colors.t,
+                    ~registration as model: model,
+                    ~font: UiFont.t,
+                    ~dispatch: msg => unit,
+                    (),
+                  ) => {
+      let%hook (loadingRotation, _animationState, _resetRotation) =
+        Hooks.animation(
+          ~name="Loading Animation",
+          loadingAnimation,
+          ~active=model.viewState == Waiting,
+        );
+
+      switch (model.viewState) {
+      | Hidden => React.empty
+      | EnteringKey =>
+        <modal theme>
+          <Component_InputText.View
+            model={model.inputModel}
+            theme
+            isFocused=true
+            fontFamily={font.family}
+            fontSize=14.
+            dispatch={msg => dispatch(InputText(msg))}
+          />
+        </modal>
+      | Waiting =>
+        <modal theme>
+          <View style=Styles.message>
+            <View
+              style=Style.[
+                width(14),
+                height(14),
+                transform([
+                  Transform.Rotate(Math.Angle.from_radians(loadingRotation)),
+                ]),
+                alignItems(`Center),
+                justifyContent(`Center),
+                marginRight(8),
+              ]>
+              <FontIcon
+                icon=FontAwesome.spinner
+                color={Colors.foreground.from(theme)}
+                fontSize=14.
+              />
+            </View>
+            <Text
+              fontFamily={font.family}
+              fontSize=14.
+              style=Styles.messageText
+              text="Validating license key..."
+            />
+          </View>
+        </modal>
+      | KeySuccess =>
+        <modal theme>
+          <View style=Styles.message>
+            <View style=Styles.messageIcon>
+              <FontIcon
+                icon=FontAwesome.checkCircle
+                fontSize=14.
+                color={Colors.foreground.from(theme)}
+              />
+            </View>
+            <Text
+              fontFamily={font.family}
+              fontSize=14.
+              style=Styles.messageText
+              text="Thank you for purchasing Onivim!"
```

Comment:
I like the flow you put together for this! Thanks for thinking through the UX here